libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/ApolloCall.kt

```diff
@@ -83,8 +85,10 @@ class ApolloCall<D : Operation.Data> internal constructor(
    *                    println("order received: ${it.data?.order?.id"})
    *                  }
    * ```
+   * @param throwApolloExceptions whether to throw [ApolloException]s (e.g. network errors, cache misses) encountered in responses,
+   * rather than emitting them in [ApolloResponse.exception]. This was the behavior in 3.x and is kept here as a convenience for migration.
    */
-  fun toFlow(): Flow<ApolloResponse<D>> {
+  fun toFlow(throwApolloExceptions: Boolean = false): Flow<ApolloResponse<D>> {
```

Comment:
Can this be on `ApolloClient.Builder()` so that one doesn't have to pass it to all `toFlow` calls?

Reply:
Yes, but I think it could be slightly confusing because `execute()` would still throw.

Reply:
Mmmm to me the current contract of `execute` is something like this:

```
  /**
   * Retrieves a single value from the given ApolloCall, ignoring any cache miss or network error
   * @throws ApolloException if the call returns zero or multiple valid GraphQL responses
   */
  suspend fun execute(): ApolloResponse<D> {
```

It could stay exactly the same in 4.x?

Reply:
I think that makes sense. I'm just wondering if a user setting `throwApolloExceptions` to `false` on ApolloClient would expect `execute()` to follow it, and thus would expect it to return an ApolloResponse with a non null exception instead of throwing? But maybe clarifying this in `throwApolloExceptions`'s KDoc is enough to avoid confusion.

Reply:
I see `throwApolloExceptions = true` as the "exceptional" case. I think we're also most likely going to need a `foldFetchExceptions`. All in all, my favorite solution would look like this:

```kotlin
/**
 * whether exceptions such as cache miss or other ApolloException should throw
 *
 * If true, the call site must catch ApolloException. This was the behaviour in Apollo Kotlin 3
 *
 * Default: false
 */
fun ApolloClient.Builder.throwOnException(throwOnException: Boolean) { }

/**
 * whether to fold the fetch exception in [CacheInfo]
 *
 * Policies such as [FetchPolicy.CacheFirst] or [FetchPolicy.NetworkFirst] are often used to retrieve a
 * single response. In such cases, set [foldFetchExceptions] to true to avoid emitting the first
 * exception (if any) and read it from [ApolloResponse.cacheInfo]?.cacheMissException and [ApolloResponse.cacheInfo]?.networkException
 *
 * Default: false
 */
fun ApolloClient.Builder.foldFetchExceptions(foldFetchExceptions: Boolean) { }

/**
 * Retrieves a single value from the given ApolloCall, ignoring any cache miss or network error
 * @throws ApolloException if the call returns zero or multiple valid GraphQL responses
 */
suspend fun ApolloCall.execute(): ApolloResponse<D> { }
```

If you want 3.x behaviour, set `throwOnException` and `foldFetchExceptions` to true. We can certainly make them deprecated from day 1 on v4. `execute()` could stay as a handy way to get a single value (also because it's not too complicated to keep around). `ThrowingFetchPolicyInterceptors.kt` could go away. WDYT?

Reply:
Yes, that looks good to me 👍